<launch>

  <arg name="skip_frames" default="1"/>

  <arg name="namespace" default="/dvrk/"/> 
  <arg name="psm" default="PSM2"/>
  <arg name="camera_registration" default="$(find dvrk_vision)/defaults/registration_params_vu.yaml" />

  <arg name="camera_parent" default="world"/>
  <arg name="transform_yaml" default="package://dvrk_vision/defaults/registration_params_vu.yaml"/>

  <arg name="width" default="1600" />
  <arg name="height" default="600" />

  <include file="$(find dvrk_vision)/launch/just_cams.launch">

    <arg name="width" default="$(arg width)" />
    <arg name="height" default="$(arg height)" />

    <arg name = "device_left" default ="/dev/video0" />
    <arg name = "device_right" default ="/dev/video1" />

    <arg name="single_image_from_camera" default="true" />

<!--     <arg name="image_format" default="raw"/>
    <arg name="format" value="YUYV"/> -->
    <arg name="image_format" default="compressed"/>
    <arg name="format" default="jpeg"/>

    <arg name="camera_info_url_left" default="file://$(find dvrk_vision)/defaults/eAP870_calibration_left_$(arg width)x$(arg height).yaml" />
    <arg name="camera_info_url_right" default="file://$(find dvrk_vision)/defaults/eAP870_calibration_right_$(arg width)x$(arg height).yaml"/>
    
    <arg name="transform_yaml" default="$(arg camera_registration)"/>
    <arg name="camera_parent" default="$(arg psm)_SIM_psm_base_link"/>
    <arg name="fps" default="30" />
    <arg name="skip_frames" default="0"/>
    <arg name="exposure" default="100" />
    <arg name="white_balance" default="4800" />
    <arg name="auto_white_balance" default="0" />
    <arg name="auto_exposure" default="1"/>
    <arg name="saturation" default="100" />
    <arg name="brightness" default="0" />
    <arg name="hue" default="-180" />
  </include>

  <node ns="stereo" pkg="nodelet" type="nodelet" name="disparity"
        args="load stereo_image_proc/disparity standalone_nodelet"
        output="screen">
    <param name="queue_size" value="5" />
    <param name="approximate_sync" value="true" />
  </node>

  <node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load /stereo/disparity $(find dvrk_vision)/defaults/stereo_processing.yaml" />

  <node ns="stereo" pkg="nodelet" type="nodelet" name="point_cloud"
        args="load dvrk_vision/point_cloud2 standalone_nodelet">
    <param name="queue_size" value="5" />
    <remap from="left/image_rect_color" to="left/image_rect"/>
    <remap from="right/image_rect_color" to="right/image_rect"/>
    <param name="approximate_sync" value="true" />
  </node>
  
</launch>